lib/cartopy/trace.pyx

```diff
@@ -249,22 +196,14 @@ cdef enum State:
     POINT_NAN
 
 
-cdef State get_state(const Point &point, const GEOSPreparedGeometry *gp_domain,
-                     GEOSContextHandle_t handle):
+cdef State get_state(const Point &point, object gp_domain):
     cdef State state
-    cdef GEOSCoordSequence *coords
-    cdef GEOSGeometry *g_point
 
     if isfinite(point.x) and isfinite(point.y):
         # TODO: Avoid create-destroy
-        coords = GEOSCoordSeq_create_r(handle, 1, 2)
-        GEOSCoordSeq_setX_r(handle, coords, 0, point.x)
-        GEOSCoordSeq_setY_r(handle, coords, 0, point.y)
-        g_point = GEOSGeom_createPoint_r(handle, coords)
-        state = (POINT_IN
-                 if GEOSPreparedCovers_r(handle, gp_domain, g_point)
-                 else POINT_OUT)
-        GEOSGeom_destroy_r(handle, g_point)
+        g_point = sgeom.Point((point.x, point.y))
+        state = POINT_IN if gp_domain.covers(g_point) else POINT_OUT
+        del g_point
```

Comment:
To avoid this create-destroy Point overhead, shapely could have some predicates specialized for taking x, y coordinates instead of a Point. 
Coincidentally, GEOS recently added this to their C API as well, so I experimented with this a bit in shapely: https://github.com/shapely/shapely/pull/1548 (and this improvement wouldn't be tied to the GEOS version).

With that, if the user has shapely 2.0 installed, the above could be replaced with:

```
state = POINT_IN if shapely.intersects_xy(gp_domain, point.x, point.y) else POINT_OUT
```

(intersects and covers should be the same in case of polygon/point predicate)

Testing this, this seems to give around 15-20% speedup on the `time_project_linear` benchmark (on my laptop, going from around 6s to 5s)

Reply:
That'd be nice to have.

Reply:
So this actually made it into shapely 2.0, and thus could be used (conditionally on the shapely version, if you are OK with adding some different code branches depending on the shapely version)